### PR TITLE
search XDG_CONFIG_HOME for myclirc

### PR DESCRIPTION
**Pull request by iTakeshi**
## Description
Search for a config file named `$XDG_CONFIG_HOME/mycli/myclirc`, as mentioned in #581 .
If `$XDG_CONFIG_HOME` is not exported, it will be defaulted to `~/.config`.

## Checklist

- [x] I've added this contribution to the `changelog.md`.
- [x] I've added my name to the `AUTHORS` file (or it's already there).

### CODE REVIEW

File: mycli/main.py [hunk 1]
```diff
@@ -91,6 +91,8 @@ class MyCli(object):
 
     system_config_files = [
         '/etc/myclirc',
+        os.path.join(
+            os.environ.get("XDG_CONFIG_HOME", "~/.config"), "mycli", "myclirc")
     ]
```
Comment (laixintao) [inline mycli/main.py]: Is it necessary to check if the `XDG_CONFIG_HOME` is valide?

For example, what if the env exist but is empty string?

Comment (iTakeshi) [reply mycli/main.py]: Yes, it has to be checked. Thanks for your comment.
I'll push a new commit.

Comment (iTakeshi) [reply mycli/main.py]: And, I don't think we don't need to check if the `XDG_CONFIG_HOME` is REALLY valid or not.
In the latest commit 8aa2466 , I added a code to check just it is empty or not.
However, when it exists and not empty, even if the referred path is NOT valid (i.e., path not exists), I think we should respect they set the very path for the envvar.

**Comment by iTakeshi** [comment]
Sorry, something went wrong. Wait for an update.

**Comment by iTakeshi** [comment]
↑ It was my simple misunderstanding. Everything works just fine at the latest commit 6d87744 .

**Comment by amjith** [comment]
Looks good. Thank you @iTakeshi 

:cookie:

**Pull request merged**